### PR TITLE
Bluetooth.requestDevice() update the options

### PR DESCRIPTION
**Pull request by hamishwillee**
This updates `Bluetooth.requestDevice()` which has many new options.

The information largely comes from https://github.com/WebBluetoothCG/web-bluetooth/issues/620 and https://developer.chrome.com/docs/capabilities/bluetooth

Note that I have not tried to update the examples to show new options. But the key ones were already covered, and the new ones are well covered in the linked document.

This follows on from #32533

### CODE REVIEW

**Comment by hamishwillee** [comment]
@chrisdavidmills In case you want to look at this one too, I've requested you as a reviewer.

This primarily updates `Bluetooth.requestDevice()` to match the current set of supported options. In the spec this is a bunch of dictionaries, but I have followed normal approach of making them an object. 

While I was here I added an event. There are others that may or may not be implemented, but I couldn't easily prove it, and I think I've gone far enough down the rabbit hole. A lot of this stuff is behind preferences and still evolving. It isn't clear what will finally fall out of it - probably not worth putting to much more into it until cannonical answers about what is being released come out.

A question though about where things are implemented. The IDL shows handlers for events such as `characteristicvaluechanged` on a bunch of interfaces, such as `Bluetooth`, `[BluetoothRemoteGATTCharacteristic](https://webbluetoothcg.github.io/web-bluetooth/#bluetoothremotegattcharacteristic)` etc. However [the event doc](https://webbluetoothcg.github.io/web-bluetooth/#event-types) says

> `characteristicvaluechanged`
>    Fired on a BluetoothRemoteGATTCharacteristic when its value changes, either as a result of a read request , or a value change notification/indication.

So my understanding is that this would be documented on the interface it is fired on - i.e.  BluetoothRemoteGATTCharacteristic - is that right?

And the fact that the event handlers are defined on other interfaces just means it might bubble?

Or to put it another way, I don't have to define this event on everything that has a handler defined do I?

**Review by chrisdavidmills — APPROVED** [review]
@hamishwillee looking pretty good. I've left a few comments for you to mull over, but only really small stuff. Approving so you can merge when done with them.

File: files/en-us/web/api/bluetooth/index.md [hunk 1]
```diff
@@ -26,6 +26,13 @@ _Inherits properties from its parent {{domxref("EventTarget")}}._
 - {{domxref("Bluetooth.requestDevice","Bluetooth.requestDevice()")}} {{Experimental_Inline}}
   - : Returns a {{jsxref("Promise")}} to a {{domxref("BluetoothDevice")}} object with the specified options.
 
+## Events
+
+Listen to these events using {{domxref("EventTarget.addEventListener", "addEventListener()")}} or by assigning an event listener to the `oneventname` property of this interface.
+
+- {{DOMxRef("Bluetooth/availabilitychanged_event", "availabilitychanged")}}
+  - : Fired on {{domxref("navigator.bluetooth")}} when the Bluetooth system as a whole becomes available or unavailable to the user agent.
+
```
Comment (hamishwillee) [inline files/en-us/web/api/bluetooth/index.md:35]: FYI According to Vinyl, this is not implemented. I don't know why the tests appear to show it is. It is extremely annoying, but even though Chrome is driving the spec, and is the sole implementer, it still doesn't match the spec.

```suggestion
```

**Comment by hamishwillee** [comment]
Thanks very much @chrisdavidmills - very happy with the review, as usual. I've removed the added availability event as apparently not present, despite appearing to work in my test. I'm not arguing with BCD.

**Pull request merged**